src/main/java/uk/gov/register/util/HashValue.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.register.util;
+
+import uk.gov.register.core.HashingAlgorithm;
+
+public class HashValue {
+    private final String value;
+    private final String hashingAlgorithm;
+
+    public HashValue(HashingAlgorithm hashingAlgorithm, String value) {
+        this.hashingAlgorithm = hashingAlgorithm.toString();
+        this.value = value;
+    }
+
+    public String encode() {
+        return hashingAlgorithm + ":" + value;
+    }
+
+    public String getValue() {
+        return value;
+    }
+
+    public static HashValue decode(String hashingAlgorithm, String encodedHash) {
+        if (!encodedHash.startsWith(hashingAlgorithm)) {
+            throw new RuntimeException("Hash cannot be decoded");
+        }
+
+        String[] parts = encodedHash.split(hashingAlgorithm + ":");
+
+        if (parts.length != 2) {
+            throw new RuntimeException(String.format("Cannot create HashValue from encoded hash %s", encodedHash));
+        }
+
```

Comment:
You do not check for the situation where you get `sha-256abcde:fghijk` - in this case parts[1] would be `fghijk` when this should not have parsed. I think you should check that the `encodeHash` starts with `sha-256:` rather than checking that it starts with `sha-256`.

Also can you throw a more specific exception rather than `RuntimeException`? I think it should also be a checked exception (rather than unchecked) so that we know in any code that calls this that we need to deal with the situation where the encodedHash cannot be decoded.